pax-chassis-web/src/lib.rs

```diff
@@ -101,28 +102,31 @@ impl PaxChassisWeb {
             .unwrap()
             .dyn_into::<HtmlCanvasElement>()
             .unwrap();
-        let context: web_sys::CanvasRenderingContext2d = canvas
-            .get_context("2d")
-            .unwrap()
-            .unwrap()
-            .dyn_into::<web_sys::CanvasRenderingContext2d>()
-            .unwrap();
-
         let width = canvas.offset_width() as f64 * dpr;
         let height = canvas.offset_height() as f64 * dpr;
-
         canvas.set_width(width as u32);
         canvas.set_height(height as u32);
-        let _ = context.scale(dpr, dpr);
-
-        let render_context = WebRenderContext::new(context, window.clone());
+        let config = RenderConfig::new(
+            false,
+            canvas.offset_width() as u32,
+            (canvas.offset_height() as u32).max(1),
+            dpr as u32,
+        );
+        let render_backend = RenderBackend::to_canvas(canvas, config)
+            .await
+            .expect("couldn't create render_backend");
 
+        let render_context = WgpuRenderer::new(render_backend);
         self.drawing_contexts.insert(id, render_context);
     }
 
-    pub fn send_viewport_update(&mut self, width: f64, height: f64) {
+    pub fn send_viewport_update(&mut self, width: f64, height: f64, dpr: f64) {
         self.engine.borrow_mut().set_viewport_size((width, height));
+        for rc in self.drawing_contexts.values_mut() {
+            rc.resize(width as f32, height as f32, dpr as f32);
```

Comment:
I don't think we can always resize all the RC's to the full viewport since some could be inside scrollers.